script.js

```diff
@@ -1,4 +1,185 @@
+    console.log("Error, input is not 1 or 2");
+    chooseDiceMessage =
+      "Please key in '1' or '2' only to choose your dice order. <br>Your dice one is: " +
+      diceRollResults[0] +
+      " and your dice two is: " +
+      diceRollResults[1];
+    return chooseDiceMessage;
+  }
+  //if player is player 1
+  if (playerCount == 1) {
+    //if order of dice is 1
+    if (inputDice == 1) {
+      twodigit = "" + Number(diceRollResults[0]) + Number(diceRollResults[1]);
+    }
+    //if order of dice is 2
+    if (inputDice == 2) {
+      var twodigit =
+        "" + Number(diceRollResults[1]) + Number(diceRollResults[0]);
+    }
+    //twodigit = 55; ** Add as control to test for tie scores
+    allPlayerResults.push(twodigit);
+    gameMode = gameMode_ROLL_DICE;
+    chooseDiceMessage =
+      "Player " +
+      playerCount +
+      ", you chose Dice " +
+      inputDice +
+      " first. <br> Your score is " +
+      twodigit +
+      ". <br><br> It is now Player 2's turn! <br>Player 2, click 'Submit!' to roll your dice!";
+    playerCount = 2;
+    diceRollResults = [];
+    return chooseDiceMessage;
+  }
+
+  //if player is player 2
+  if (playerCount == 2) {
+    var twodigit = "";
+    if (inputDice == 1) {
+      twodigit = "" + Number(diceRollResults[0]) + Number(diceRollResults[1]);
+    }
+    //if order of dice is 2
+    if (inputDice == 2) {
+      twodigit = "" + Number(diceRollResults[1]) + Number(diceRollResults[0]);
+    }
+    //twodigit = 55; ** Add as control to test for tie scores
+    allPlayerResults.push(twodigit);
+    chooseDiceMessage =
+      "Player " +
+      playerCount +
+      ", You chose Dice " +
+      inputDice +
+      " first.<br> Your score is " +
+      twodigit +
+      ". <br><br> Click 'Submit!' to compare your scores!";
+    gameMode = gameMode_COMPARE_SCORES;
+    diceRollResults = [];
+    return chooseDiceMessage;
+  }
+  return chooseDiceMessage;
+};
+
+///////////////////////////////////
+// 4. COMPARE PLAYER SCORES
+///////////////////////////////////
+var compareScores = function () {
+  //Player 1 score is higher
+  if (allPlayerResults[0] > allPlayerResults[1]) {
+    return (
+      "Player 1: " +
+      allPlayerResults[0] +
+      "<br> Player 2: " +
+      allPlayerResults[1] +
+      "<br><br> Player 1 wins with a total score of: " +
+      allPlayerResults[0] +
+      "! <br>Click 'Submit' to play again!"
+    );
+  }
+  //Player 2 score is higher
+  if (allPlayerResults[1] > allPlayerResults[0]) {
+    return (
+      "Player 1: " +
+      allPlayerResults[0] +
+      "<br> Player 2: " +
+      allPlayerResults[1] +
+      "<br><br> Player 2 wins with a total score of: " +
+      allPlayerResults[1] +
+      "! <br>Click 'Submit' to play again!"
+    );
+  }
+  // If scores are tied
+  return (
+    "Player 1: " +
+    allPlayerResults[0] +
+    "<br> Player 2: " +
+    allPlayerResults[1] +
+    "<br><br> It's a tie! <br>Click 'Submit' to play again!"
+  );
```

Comment:
good understand of implicity where you know that whatever is left must be the `tie` situation. Try to apply this concept to the rest of your `if` statements as well!